states/huntsman/sleeping.py

```diff
@@ -4,8 +4,11 @@ def on_enter(event_data):
     pocs.next_state = 'ready'
 
     # If it is dark and safe we shouldn't be in sleeping state
-    if pocs.is_dark() and pocs.is_safe():
-        pocs.say("Weather is good and it is dark. Something must have gone wrong. Stopping loop.")
-        pocs.stop_states()
+    if pocs.is_safe(horizon='focus'):
+        msg = "Weather is good and it is dark. Something must have gone wrong."
+        pocs.say(msg)
+        if not pocs.should_retry:
+            pocs.stop_states()
+            raise RuntimeError(msg)
```

Comment:
```suggestion
    if pocs.is_safe(horizon='focus'):
        pocs.say("Weather is good and it is dark. Something must have gone wrong.")
        pocs.stop_states()
````


The `exit_when_done=True` param on the `POCS` constructor will cause the machine to exit gracefully after it has been stopped via `stop_states`. You probably just want to set that on your `pocs` instance. If we always want it to exit in that situation then can just change that default.